feat: add manual-revocation-check script (#177)
mattrglobal/sample-apps: manual-revocation-check/README.md

Reviewer: Do you think we could/should add an example credential of each type for them to play with?

Also do we want to show what the result of that script looks like or explain what it actually does?

Author: A JSON credential example is basically there already (though we only share the `credentialStatus` object, not the whole credential). We certainly can add a compact credential as well.

The output is just text in the terminal, but I can add a section to give a sample output.

[![MATTR](../docs/assets/mattr-logo-square.svg)](https://github.com/mattrglobal)

# Manual revocation check

## Overview

Credential revocation is a critical feature that allows issuers to invalidate previously issued credentials, ensuring the integrity and security of the system.

> MATTR provides capabilities to issue revokable credentials, to manage the revocation status of credentials, and to check the revocation status of a credential. Check out our [Credential Management](https://learn.mattr.global/docs/capabilities/management) capabilities and our [Revocation Tutorial](https://learn.mattr.global/tutorials/management/revocation) to learn more.

Reviewer: ```suggestion
MATTR provides capabilities to issue revokable credentials, to manage the revocation status of credentials, and to check the revocation status of a credential. Check out our [credential management](https://learn.mattr.global/docs/capabilities/management) capabilities and our [Revocation tutorial](https://learn.mattr.global/tutorials/management/revocation) to learn more.
```

> While we highly suggest to leverage your MATTR tenant for credential management, credentials also include the necessary information to check their revocation status manually to enable interoperability.

Reviewer: ```suggestion
While we highly recommend to leverage your MATTR tenant for credential management, credentials also enable unauthenticated relaying parties to check their revocation status to enable interoperability.
```

Author: ```suggestion
While we highly recommend to leverage your MATTR tenant for credential management, credentials also enable unauthenticated relying parties to check their revocation status to enable interoperability.
```

This sample script demonstrates how to retrieve the revocation list data, how to decode the data, and how to check the revocation status.

## How to run the script

> ### Web Semantic Credentials

Reviewer: ```suggestion
### JSON credentials
```

> Web Semantic credentials contain a `credentialStatus` field that holds the reference to a revocation list.

Reviewer: ```suggestion
JSON credentials contain a `credentialStatus` field that holds the reference to a revocation list.
```

```json
    "credentialStatus": {
        "id": "https://re-labs.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/0ec79c8e-9859-46c0-a277-6e48f468b16e#1",
        "type": "RevocationList2020Status",
        "revocationListIndex": "1",
        "revocationListCredential": "https://re-labs.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/0ec79c8e-9859-46c0-a277-6e48f468b16e"
    },
```

You can check the revocation status of this credential by running the script with node, providing the `revocationListCredential` and `revocationListIndex` as arguments.

```bash
node manual-revocation-check.js https://learn.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/0ec79c8e-9859-46c0-a277-6e48f468b16e 1
```

> ### Compact and Compact-Semantic Credentials

Reviewer: ```suggestion
### CWT and semantic CWT credentials
```

> Compact and compact-semantic credentials are usually shared and verified in their encoded forms. For convenience, you can provide the encoded credential or the revocation list URL and index to the script.

Reviewer: ```suggestion
CWT and Semantic CWT credentials are usually shared and verified in their encoded forms. For convenience, you can provide the encoded credential or the revocation list URL and index to the script.
```

> If you provide the encoded compact credential, the script will decode the credential and retrieve the list URL and index.

Reviewer: ```suggestion
If you provide the encoded form, the script will decode the credential and retrieve the list URL and index.
```

```bash
node manual-revocation-check.js CSC:/1/2KCE3I...
```

> If you decode the compact credential yourself, you can pass the revocation list URL and index.

Reviewer: ```suggestion
If you decode the credential yourself, you can pass the revocation list URL and index.
```

This sample script demonstrates how to retrieve the revocation list data, how to decode the data, and how to check the revocation status.

## How to run the script

### JSON credentials

JSON credentials contain a `credentialStatus` field that holds the reference to a revocation list.

```json
    "credentialStatus": {
        "id": "https://labs-mattr-university.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/57698edc-0826-4628-a2eb-888cb840d4b5#0",
        "type": "RevocationList2020Status",
        "revocationListIndex": "0",
        "revocationListCredential": "https://labs-mattr-university.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/57698edc-0826-4628-a2eb-888cb840d4b5"
    },
```

You can check the revocation status of this credential by running the script with node, providing the `revocationListCredential` and `revocationListIndex` as arguments.

```bash
node manual-revocation-check.js https://learn.vii.au01.mattr.global/core/v2/credentials/web-semantic/revocation-lists/0ec79c8e-9859-46c0-a277-6e48f468b16e 1
```

> ### CWT and semantic CWT credentials

Reviewer: ```suggestion
### CWT and Semantic CWT credentials
```
